### ISSUE TITLE
[Feature Request]: More extensions

### ISSUE THREAD

**Issue by nam-co** [beyondcode/herd-community#16]
### Feature Description

Instead of opening an issue for each extension we wish, easier like this

**Comment by nam-co** [comment]
FFMPEG

**Comment by nam-co** [comment]
puppeteer: Useful for packages like browsershot (amazon package)

**Comment by rafaelgrube** [comment]
I often use the oci8 extension. It would be awesome if I can add other extensions.

**Comment by pyrou** [comment]
Would be definitely nice to let us pick which extension we would like to enable.

At least provide a per-php-version php.ini file, so advanced user can add their own extension=/absolute/path.so, because so far, since the php.ini file is shared across all version, no chance to add an extension that would work for multiple php version.

**Comment by ryanmortier** [comment]
sqlsrv / pdo_sqlsrv

SQL Server is a database that has first-party support through Laravel.

**Comment by renedekat** [comment]
xdebug

A debugger is an essential tool during development.

**Comment by grantholle** [comment]
ldap

**Comment by mpociot** [comment]
Taken from #47  and #26 
* GRPC
* memcached

Right now, we'll just gather feedback for additional extensions and then see if/which ones we can bundle into Herd.

**Comment by bradgenereux** [comment]
Ghostscript: Imagick requires it for PDF operations.

**Comment by eliurkis** [comment]
> `sqlsrv` `pdo_sqlsrv`
> 
> SQL Server is a database that has first-party support through Laravel.

🙏

**Comment by ryanscherler** [comment]
It seems that `mbregex` isn't explicitly compiled with the included `mbstring` extension and causing issues with running Craft CMS via Herd (had to revert back to Homebrew 😔). Anybody else experience the same issue?

**Comment by mpociot** [comment]
> It seems that `mbregex` isn't explicitly compiled with the included `mbstring` extension and causing issues with running Craft CMS via Herd (had to revert back to Homebrew 😔). Anybody else experience the same issue?
> 
> 
> 
> The error I get when running any version of Herd PHP:
> 
> 
> 
> ```php
> 
> error \mb_regex_encoding('UTF-8')
> 
> 
> 
> // call to undefined function mb_regex_encoding()
> 
> ```

Please make sure that you've updated the built in PHP version via the settings screen.

**Comment by ryanscherler** [comment]
Oh ok - has this extension been enabled and not addresses this error?

**Comment by Plytas** [comment]
> Taken from #47  and #26 
> * GRPC
> * memcached
> 
> Right now, we'll just gather feedback for additional extensions and then see if/which ones we can bundle into Herd.

Protobuf too.

**Comment by vincent-tarrit** [comment]
Hello ! Thank you for the app ! It looks really nice ! I'm just waiting to download because I would need the extension 

IMAP

To be sure to use in all my project 😃 

Thank you and I hope it will be added 
Vincent

**Comment by ewilan-riviere** [comment]
I love Herd! 😄

What about pcov for tests coverage?

**Comment by mpociot** [comment]
@ewilan-riviere I'm currently working on compiling xDebug for all PHP versions so that we can ship it with Herd, which you can then use for test coverage 👍

**Comment by alcaeus** [comment]
@mpociot is there any progress on the MongoDB support? Anything we can help with in that regard?

**Comment by pryley** [comment]
> @ewilan-riviere I'm currently working on compiling xDebug for all PHP versions so that we can ship it with Herd, which you can then use for test coverage 👍

[pcov](https://github.com/krakjoe/pcov) is almost 5 times faster than xdebug for code coverage.

> The differences in performance of Xdebug and PCOV are not slight. Xdebug is first and foremost a debugging extension, and when you load it, you incur the overhead of a debugger even when it's disabled. PCOV is less than 1000 lines of code (not including CFG) and doesn't have anything like the overhead of a debugger.

**Comment by BurakBoz** [comment]
Can you add opcache extension and gd extension doesn't have avif support yet is this two hard to build statically?

**Comment by esseremmerik** [comment]
We use php_xsl extension, would love to see this included

**Comment by Plytas** [comment]
I believe the best approach here would be to provide tooling to build our own binaries. A default binary with most popular extensions is always good, but people will constantly ask for more.

**Comment by ianlandsman** [comment]
A second vote for the IMAP extension if it's possible.

**Comment by nyeyint** [comment]
Is there any workaround to enable the **oci8** extension for now?

**Comment by jacobshenning** [comment]
@mpociot Have you looked into compiling openswoole at all? It would be awesome if that was a possibility.

**Comment by ndberg** [comment]
php-decimal

Correctly-rounded, arbitrary precision decimal floating-point arithmetic in PHP 

[https://php-decimal.io/#introduction](https://php-decimal.io/#introduction)
[https://github.com/php-decimal/ext-decimal](https://github.com/php-decimal/ext-decimal)

**Comment by morrislaptop** [comment]
'swoole'

Swoole provides concurrency for PHP with Laravel Octane.

From #97

**Comment by jacobshenning** [comment]
Have any decisions been made to not implement an extension? If so, is there a way I can see the won't implement list?

**Comment by vikas5914** [comment]
@pryley You can download the Pre-complied extension from here: https://packages.macports.org/php82-pcov/

@morrislaptop  Swoole: https://packages.macports.org/php82-swoole/

**Comment by stojankukrika** [comment]
Mongo db will be nice to have as well.

**Comment by vikas5914** [comment]
@stojankukrika You can download from here: https://packages.macports.org/php82-mongodb/

**Comment by chrisbaswell** [comment]
Snowflake PDO - https://github.com/snowflakedb/pdo_snowflake

**Comment by StefanNeuser** [comment]
mcrypt - please :-)

**Comment by iambateman** [comment]
gettext. 

I _think_ switching to herd made something like `__('Translatable')` break, and as far as I can tell, there's no way to enable it within Herd.

**Comment by nam-co** [comment]
I haven't found any issues with json translations on any project, maybe is something else

**Comment by thijsdewitt** [comment]
`ext-xsl` Would be nice for us Magento developers

**Comment by milhouse1337** [comment]
I've compiled a list of extensions that were included in the brew build but are absent in this one. Here's the breakdown, along with my reasoning for their potential inclusion:

- `gettext`: This is widely used for translation purposes and is not available via PECL.
- `igbinary`: This is very useful, although it can be installed separately.
- `memcached`: This is also very useful and can be installed via PECL.
- `msgpack`: It is very useful and can be installed via PECL.
- `pcntl`: This is used for low-level process interactions and is not available via PECL.
- `xsl`: This is used for XSL transformations and is not available via PECL.

In summary, I believe we should at least include `gettext`, `pcntl`, and `xsl`, given that there aren't alternative installation methods for them via PECL.

**Comment by pyrou** [comment]
Pecl still requires to enable (manually or not) extension in php.ini and problem is: the php.ini file is shared across all php versions in herd.

**Comment by martinemmert** [comment]
Can we pay or support you otherwise to include the proposal of @milhouse1337 ? 

I am currently stuck building an MVP and need XSL badly.
Tinker around and lose more time with `phpize` is something that I really want to avoid.

You did an awesome job and provided us Herd for free.
Saving me a ton of time is worth my money 👍

**Comment by pryley** [comment]
@martinemmert Use Valet?

**Comment by martinemmert** [comment]
@pryley never used it before and Herd was as easy as it can get setting up a php environment. 

It just works and I don't have to care about it that much. 

Nevertheless, I'll take a look at Valet. 

Thanks

**Comment by hans0801** [comment]
> @pyrou ,
> 
> With Herd's latest build, there are now php.ini files for each version of PHP. So you can add extensions per version of PHP.

Hi @jason-beach , 
Thanks fo the information. 
So the new released version had every ini file right? 
Actually i need ldap extension on this herd, could you help me for add those extension when i will use herd? 
Thanks

**Comment by vikas5914** [comment]
@martinemmert : Go to this link: https://packages.macports.org/php82-xsl/ (you can php version) and download the version you want (arm64 for M1 mac), and it will give the `.so` file that you can directly put in herd directory and change the php.ini file.

**Comment by esseremmerik** [comment]
The solution by @vikas5914 is perfect! Don't forget to open the extension file in the finder with a random application and allow the downloaded file from the internet (unknown supplier), otherwise PHP is not able to load the extension (if you have enabled strict file restrictions in OSX)

**Comment by baskinvolkan** [comment]
> `sqlsrv` 
> 
> SQL Server is a database that has first-party support through Laravel.

🙏
